### PR TITLE
Flask: add GET /api/datasets/:id

### LINKED ISSUE #76: GET /api/datasets/:id
Retrieve just the dataset with the specified primary key, but also include an array of analyses it is involved in and data for the owning tissue sample.

### PR DESCRIPTION
**Pull request by Madelinehazel**
closes #76

### CODE REVIEW

**Review by kevinlul — CHANGES_REQUESTED** [review]
For GET `/api/datasets/:id`, it shouldn't just return the same thing as the main one. Since we're fetching an individual object, it should just return that object rather than a single-element array. There were also some additional requirements in the issue that make this endpoint slightly different -- joining with analyses and all metadata for the tissue sample.

**Comment by kevinlul** [comment]
Can I take over this branch to incorporate changes from #91 ?

**Comment by Madelinehazel** [comment]
> Can I take over this branch to incorporate changes from #91 ?

Go for it

File: flask/app/routes.py [hunk 1]
```diff
@@ -356,6 +370,34 @@ def datasets_list():
...
+@app.route('/api/datasets/<int:id>', methods=['GET'])
+@login_required
+def get_dataset(id: int):
+    dataset = db.session.query(models.Dataset).filter_by(dataset_id=id).options(
+        joinedload(models.Dataset.analyses),
+        joinedload(models.Dataset.tissue_sample).
+        joinedload(models.TissueSample.participant).
+        joinedload(models.Participant.family)
+     ).one()
+    if not dataset:
```
Comment (delvinso) [inline flask/app/routes.py:382]: What are the conditions for this? Requesting an ID not found in the table doesn't trigger this as it throws an error on the query (`sqlalchemy.orm.exc.NoResultFound: No row was found for one()`)

Comment (kevinlul) [reply flask/app/routes.py:382]: I was using the wrong method -- should be fixed now.

File: flask/app/routes.py [hunk 2]
```diff
@@ -356,6 +370,34 @@ def datasets_list():
...
+@login_required
+def get_dataset(id: int):
+    dataset = models.Dataset.query.filter_by(dataset_id=id).options(
+        joinedload(models.Dataset.analyses),
+        joinedload(models.Dataset.tissue_sample).
+        joinedload(models.TissueSample.participant).
+        joinedload(models.Participant.family)
+     ).one_or_none()
+    if not dataset:
+        return 'Not Found', 404
```
Comment (delvinso) [inline flask/app/routes.py:383]: Could this return the ID that wasn't found or would it be exposing too much information?

Comment (kevinlul) [reply flask/app/routes.py:383]: The ID is in the URL parameter `/api/datasets/123413`

Comment (delvinso) [reply flask/app/routes.py:383]: That's true, was thinking it would be more for confirmation that the ID requested was the one actually queried

**Review by delvinso — APPROVED** [review]

**Pull request merged**